Author: John Beshir
Patch to disable xlib assertions, preventing sloppy programs from being automatically terminating, fixing some that had previously worked.
--- libxcb-1.0/src/xcb_xlib.c	2006-11-03 01:38:00.000000000 +0000
+++ libxcb-1.0/src/xcb_xlib.c.new	2008-02-17 10:17:08.000000000 +0000
@@ -38,7 +38,6 @@
 void xcb_xlib_lock(xcb_connection_t *c)
 {
     _xcb_lock_io(c);
-    assert(!c->xlib.lock);
     c->xlib.lock = 1;
     c->xlib.thread = pthread_self();
     _xcb_unlock_io(c);
@@ -47,8 +46,6 @@
 void xcb_xlib_unlock(xcb_connection_t *c)
 {
     _xcb_lock_io(c);
-    assert(c->xlib.lock);
-    assert(pthread_equal(c->xlib.thread, pthread_self()));
     c->xlib.lock = 0;
     pthread_cond_broadcast(&c->xlib.cond);
     _xcb_unlock_io(c);
